File: server/mvc/routes/posts.js
const { Router } = require("express");
const {
    deletePost,
    getComments,
    getPost,
    getPosts,
    getPostsCount,
    likePost,
} = require("../controllers/posts");
const { authorizeUser, putLimit, getLimit } = require("./middleware/middleware");

const router = Router();

router.post("/deletePost", authorizeUser, putLimit, deletePost);
router.post("/likePost", authorizeUser, putLimit, likePost);

router.get("/getPostsCount/:userId", getLimit, getPostsCount);
router.get("/getPosts/:page/:userId?", getLimit, getPosts);
router.get("/getPost", getLimit, getPost);
router.get("/getComments/:postId", getLimit, getComments);

module.exports = router;

File: server-go/routes/posts.go
package routes

import "github.com/gorilla/mux"

func RegisterPostsRoutes(router *mux.Router) {
	postRouter := router.PathPrefix("/posts").Subrouter()

	postRouter.HandleFunc("/deletePost", DeletePost).Methods("POST") // validations required
	postRouter.HandleFunc("/likePost", LikePost).Methods("POST")     // validations required

	postRouter.HandleFunc("/getPostsCount/{userId}", GetPostsCount).Methods("GET")     // no validation
	postRouter.HandleFunc("/getPosts/{page:[0-9]+}", GetPosts).Methods("GET")          // no validation
	postRouter.HandleFunc("/getPosts/{page:[0-9]+}/{userId}", GetPosts).Methods("GET") // no validation
	postRouter.HandleFunc("/getPost", GetPost).Methods("GET")                          // no validation
	postRouter.HandleFunc("/getComments/{postId}", GetComments).Methods("GET")         // no validation
}
